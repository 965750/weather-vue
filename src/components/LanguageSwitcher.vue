<template>
    <div class="lang d-flex text-center text-uppercase">
        <div
            v-for="language in languages"
            :key="language"
            :class="{lang__btn: true, 'lang__btn--active': getLanguage === language}"
            @click="lang = language"
        >
            {{ language }}
        </div>
    </div>
</template>

<script>
import { mapGetters } from 'vuex';

export default {
  name: 'LanguageSwitcher',
  data() {
    return {
      lang: '',
      languages: ['pl', 'en'],
    };
  },
  computed: {
    ...mapGetters(['getLanguage']),
  },
  mounted() {
    this.lang = this.getLanguage;
  },
  watch: {
    lang(val) {
      this.$i18n.locale = val;
      this.$store.dispatch('changeLanguage', val);
    },
  },
};
</script>

<style lang="scss" scoped>
@import '../assets/main';

.lang {
    width: 70px;
    border-radius: 30px;
    height: 30px;
    overflow: hidden;
    color: #FFF;
    position: absolute;
    top: 10px;
    right: 5px;

    &__btn {
        width: 50%;
        background: $deepCerulean;
        line-height: 30px;
        cursor: pointer;
        font-size: 14px;

        &:hover {
            background: lighten($deepCerulean, 5%);
        }

        &--active {
            background: darken($deepCerulean, 10%);
            color: darken(#FFF, 10%);

            &:hover {
                background: darken($deepCerulean, 5%);
            }
        }

        &:first-child {
            border-right: 1px solid $aluminium;
        }
    }
}
</style>
