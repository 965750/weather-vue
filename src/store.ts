import Vue from 'vue';
import Vuex from 'vuex';
import axios from 'axios';
import config from '@/config';

Vue.use(Vuex);

export default new Vuex.Store({
  state: {
    lang: 'en',
    card: null,
  },
  mutations: {
    setCard(state, data) {
      const weatherList = data.list.map((weather: any) => ({
        clouds: weather.clouds.all,
        time: weather.dt,
        wind: weather.wind.speed,
        main: {
          humidity: weather.main.humidity,
          pressure: weather.main.pressure,
          temp: weather.main.temp,
        },
      }));

      state.card = {
        city: {
          name: data.city.name,
          country: data.city.country,
        },
        weatherList,
      };
    },
  },
  actions: {
    citySearch({ commit }, city: string) {
      axios
        .get(`https://api.openweathermap.org/data/2.5/forecast?q=${city}&APPID=${config.apiKey}`)
        .then((res) => {
          commit('setCard', res.data);
        });
    },
  },
});
