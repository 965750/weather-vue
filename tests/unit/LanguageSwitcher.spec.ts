import {
  shallowMount,
  createLocalVue,
} from '@vue/test-utils';
import Vuex from 'vuex';
import VueI18n from 'vue-i18n';
import LanguageSwitcher from '@/components/LanguageSwitcher.vue';
import store from '@/store';
import messages from '@/locale/languages';

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(VueI18n);

const i18n = new VueI18n({
  locale: store.getters.getLanguage,
  messages,
});

const initialStore = {
  getters: {
    getLanguage: () => 'en',
  },
};

describe('LanguageSwitcher.vue', () => {
  const createWrapper = (mockStore = initialStore) => shallowMount(LanguageSwitcher, {
    store: new Vuex.Store(mockStore),
    i18n,
    localVue,
    stubs: {
      'font-awesome-icon': true,
    },
  });

  it('should match initial snapshot', () => {
    const wrapper = createWrapper();

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should swap active button', () => {
    const wrapper = createWrapper({
      getters: {
        getLanguage: () => 'pl',
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should change current lang on click', () => {
    const wrapper = createWrapper();
    expect(wrapper.vm['_data'].lang).toBe('en');

    const btnNotActive = wrapper.find('.lang__btn');
    btnNotActive.trigger('click');

    expect(wrapper.vm['_data'].lang).toBe('pl');
  });
});
