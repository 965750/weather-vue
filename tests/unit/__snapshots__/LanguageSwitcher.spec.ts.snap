// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSwitcher.vue should match initial snapshot 1`] = `
<div
  class="lang d-flex text-center text-uppercase"
>
  <div
    class="lang__btn"
  >
    
        pl
    
  </div>
  <div
    class="lang__btn lang__btn--active"
  >
    
        en
    
  </div>
</div>
`;

exports[`LanguageSwitcher.vue should swap active button 1`] = `
<div
  class="lang d-flex text-center text-uppercase"
>
  <div
    class="lang__btn lang__btn--active"
  >
    
        pl
    
  </div>
  <div
    class="lang__btn"
  >
    
        en
    
  </div>
</div>
`;
